src/target_intacct/payroll_journal.py

```diff
@@ -0,0 +1,174 @@
+import json
+import sys
+
+import pandas as pd
+
+import singer
+
+from .utils import get_input, set_journal_entry_value
+
+logger = singer.get_logger()
+
+"""
+Uploads Financial Journals to Intacct
+
+Retrieves objects from Intacct API for verifying input data
+Calls load_entries method
+Sends entries for uploading to Intacct
+"""
+
+
+def journal_upload(intacct_client, object_name) -> None:
+    logger.info("Starting upload.")
+
+    # Load Active Classes, Customers, Accounts
+    account_ids = intacct_client.get_entity(
+        object_type="general_ledger_accounts", fields=["ACCOUNTNO"]
+    )
+    business_units = intacct_client.get_entity(
+        object_type="classes", fields=["CLASSID"]
+    )
+    location_ids = intacct_client.get_entity(
+        object_type="locations", fields=["LOCATIONID"]
+    )
+    practice_area_ids = intacct_client.get_entity(
+        object_type="departments", fields=["DEPARTMENTID"]
+    )
+
+    # Load Journal Entries CSV to post + Convert to Intacct format
+    journal_entries = load_journal_entries(
+        account_ids,
+        business_units,
+        location_ids,
+        practice_area_ids,
+        object_name,
+    )
+
+    # Post the journal entries to Intacct
+    for je in journal_entries:
+        intacct_client.post_journal(je)
+
+    logger.info("Upload completed")
+
+
+"""
+Loads inputted data into Financial Journal Entries
+"""
+
+
+def load_journal_entries(
+    account_ids,
+    business_units,
+    location_ids,
+    practice_area_ids,
+    object_name,
+):
+    # Get input from pipeline
+    input_value = get_input()
+
+    # Convert input from dictionary to DataFrame
+    data_frame = pd.DataFrame(input_value)
+    # Verify it has required columns
+    cols = list(data_frame.columns)
+    REQUIRED_COLS = [
+        "Transaction Date",
+        "Class",
+        "Account Number",
+        "Account Name",
+        "Posting Type",
+        "Description",
+    ]
+
+    if not all(col in cols for col in REQUIRED_COLS):
+        logger.error(
+            f"CSV is missing REQUIRED_COLS. Found={json.dumps(cols)}, Required={json.dumps(REQUIRED_COLS)}"
+        )
+        sys.exit(1)
```

Comment:
I know this isn't your code but this should probably raise an exception instead of `sys.exit`.